drivers/Ticker.h

```diff
@@ -155,14 +155,16 @@ class Ticker : public TimerEvent, private NonCopyable<Ticker> {
      */
     void detach();
 
+#if !defined(DOXYGEN_ONLY)
```

Comment:
This sort of change doesn't make sense, does it? If you always want to exclude this bit from Doxygen, then there shouldn't be any Doxygen comments in it. Can't you just lose the Doxygen `*<` markers, or is there a need to exclude the source too?

Reply:
@aashishc1988 @AnotherButler

Reply:
As per my understanding, the source should be excluded as well. What would be the reason to include source, if the comments are removed anyway? In general, do we usually expose private/protected functions out in the docs?

Reply:
We do not want to include functions that are not public in our rendered documentation.